xdsl/dialects/riscv.py

```diff
@@ -2527,12 +2530,19 @@ def assembly_line_args(self) -> tuple[AssemblyInstructionArg, ...]:
     def custom_parse_attributes(cls, parser: Parser) -> dict[str, Attribute]:
         attributes = dict[str, Attribute]()
         attributes["immediate"] = parse_immediate_value(parser, i32)
+        fast = FastMathFlagsAttr("none")
```

Comment:
Is this flag meant to be optional? It seems you can't parse it with the flag unset?

Reply:
As on the other PR, it's not meant to be optional, `op_prop_def` just doesn't actually use the `default` parameter at all, so parsing fails when it's set as a non-`opt` `prop_def`

Reply:
I would recommend making it non-optional, and to do the default value handling manually. What's the error when you change the declaration to `prop_def`?

Reply:
I guess this is more a question for the equivalent line below, if we're removing the LiOp changes

Reply:
```
# |     raise VerifyException(f"property {prop_name} expected")
# | xdsl.utils.exceptions.VerifyException: property fastmath expected
```
That's the error I get, even with setting a default via `fastmath = prop_def(FastMathFlagsAttr, default=FastMathFlagsAttr("none")) (which also doesn't typecheck)

Reply:
yep, the default field here doesn't work. I think @alexarice is now looking into this, but for the time being we can just set it directly in the init and in the custom parser. So all that would be necessary in this PR is to change `opt_prop_def` to `prop_def`

Reply:
I've just pushed a commit that does exactly that, but it's failing the filecheck test with the same error as above. Is there something else I need to do ? I am setting the `fastmath` property in both the parse and init methods